### ISSUE TITLE
panic: unexpected return code from pcre_exec: -8

### ISSUE THREAD

**Issue by alexisph** [openshift/must-gather-clean#87]
## Bug Report

**What did you do?**
Use `must-gather-clean` from main on a must-gather from OCP v4.10.

**What did you expect to see?**
Successful execution.

**What did you see instead? Under which circumstances?**
Receiving the below error from `must-gather-clean`, every time, after running for many hours (3.5-4 hours with 12 threads), at approximately around 80% of the cleaning process.

```
panic: unexpected return code from pcre_exec: -8

goroutine 27 [running]:
github.com/gijsbers/go-pcre.matched(0x63?)
        github.com/gijsbers/go-pcre@v0.0.0-20161214203829-a84f3096ab3c/pcre.go:418 +0x9c
github.com/gijsbers/go-pcre.(*Matcher).MatchString(0xc000d98000, {0xc001340000?, 0x7c?}, 0x6007fcf2eae5262?)
        github.com/gijsbers/go-pcre@v0.0.0-20161214203829-a84f3096ab3c/pcre.go:359 +0x33
github.com/gijsbers/go-pcre.Regexp.MatcherString({{0xc000162380, 0x63, 0x63}, {0x0, 0x0, 0x0}}, {0xc001340000, 0x66f95}, 0xc0011e6a45?)
        github.com/gijsbers/go-pcre@v0.0.0-20161214203829-a84f3096ab3c/pcre.go:299 +0xb5
github.com/openshift/must-gather-clean/pkg/obfuscator.(*regexObfuscator).replace(0xc00031f810, {0xc001340000, 0x66f95})
        github.com/openshift/must-gather-clean/pkg/obfuscator/regex.go:26 +0xae
github.com/openshift/must-gather-clean/pkg/obfuscator.(*regexObfuscator).Contents(0xc00012d980?, {0xc001340000?, 0x66f88?})
        github.com/openshift/must-gather-clean/pkg/obfuscator/regex.go:21 +0x1e
github.com/openshift/must-gather-clean/pkg/obfuscator.(*targetObfuscator).Contents(0xc0011a2000?, {0xc001340000?, 0xc00079fb30?})
        github.com/openshift/must-gather-clean/pkg/obfuscator/target_obfuscator.go:19 +0x5f
github.com/openshift/must-gather-clean/pkg/obfuscator.(*MultiObfuscator).Contents(0x6b2f78?, {0xc0011a2000?, 0xc00079fbe0?})
        github.com/openshift/must-gather-clean/pkg/obfuscator/multi_obfuscator.go:17 +0x71
github.com/openshift/must-gather-clean/pkg/cleaner.(*ContentObfuscator).ObfuscateReader(0xc0003768c0, {0x6b3318?, 0xc000186128?}, {0x6b3358?, 0xc000186130?})
        github.com/openshift/must-gather-clean/pkg/cleaner/cleaning.go:175 +0x2ef
github.com/openshift/must-gather-clean/pkg/cleaner.(*FileContentObfuscator).ObfuscateFile(0xc0003768c0, {0xc0000c05c6?, 0xbb?}, {0xc0000c05c6, 0xbb})
        github.com/openshift/must-gather-clean/pkg/cleaner/cleaning.go:126 +0x32f
github.com/openshift/must-gather-clean/pkg/cleaner.(*FileProcessor).Process(0xc0003768c0, {0xc0000c05c6, 0xbb})
        github.com/openshift/must-gather-clean/pkg/cleaner/cleaning.go:92 +0x165
github.com/openshift/must-gather-clean/pkg/traversal.(*Worker).ProcessQueue(0xc0002b2990, 0x0?, 0x0?)
        github.com/openshift/must-gather-clean/pkg/traversal/worker.go:40 +0x206
github.com/openshift/must-gather-clean/pkg/traversal.(*FileWalker).Traverse.func1(0x0?, 0x0?, 0x0?)
        github.com/openshift/must-gather-clean/pkg/traversal/traversal.go:32 +0x42
created by github.com/openshift/must-gather-clean/pkg/traversal.(*FileWalker).Traverse
        github.com/openshift/must-gather-clean/pkg/traversal/traversal.go:31 +0xd5
```

**Environment**
* must-gather-clean version:

```
Version: pkg.Version{Version:"v0.0.1-8-g8432051", GitCommit:"8432051", BuildDate:"2023-08-30T09:45:24Z", GoOs:"linux", GoArch:"amd64"}
```

* OpenShift Version:

```
Client Version: 4.12.28
Kustomize Version: v4.5.7
Server Version: 4.10.46
Kubernetes Version: v1.23.12+8a6bfe4
```

**Additional context**
Used `must-gather-clean` many times in the past with no issue. This did not happen before commit `b4da3593a76a51cc87bb20de3ddbd942597cfdcf`.
I pulled from main and build `must-gather-clean` on 2023-08-30. I had to install `pcre-devel` on RHEL 9.2 first.

**Comment by tjungblu** [comment]
I assume this is caused by the gzip files that are also reported in #88.

This will be fixed with #90 and in the next release that comes out later.

**Comment by alexisph** [comment]
This is still an issue:

```log
panic: unexpected return code from pcre_exec: -8                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          goroutine 28 [running]:                                                                                                                                                                                                                                                      github.com/gijsbers/go-pcre.matched(0x63?)                                                                                                                                                                                                                                           github.com/gijsbers/go-pcre@v0.0.0-20161214203829-a84f3096ab3c/pcre.go:418 +0x9c                                                                                                                                                                                     github.com/gijsbers/go-pcre.(*Matcher).MatchString(0xc05d6d7800, {0xc06ed14000?, 0x0?}, 0x69b5cf?)                                                                                                                                                                                   github.com/gijsbers/go-pcre@v0.0.0-20161214203829-a84f3096ab3c/pcre.go:359 +0x33                                                                                                                                                                                     github.com/gijsbers/go-pcre.Regexp.MatcherString({{0xc0001ea380, 0x63, 0x63}, {0x0, 0x0, 0x0}}, {0xc06ed14000, 0x69b5cf}, 0xc000289dc0?)                                                                                                                                             github.com/gijsbers/go-pcre@v0.0.0-20161214203829-a84f3096ab3c/pcre.go:299 +0xb5                                                                                                                                                                                     github.com/openshift/must-gather-clean/pkg/obfuscator.(*regexObfuscator).replace(0xc000321b80, {0xc06ed14000, 0x69b5cf})                                                                                                                                                             github.com/openshift/must-gather-clean/pkg/obfuscator/regex.go:26 +0xae                                                                                                                                                                                              github.com/openshift/must-gather-clean/pkg/obfuscator.(*regexObfuscator).Contents(0xc0001a9920?, {0xc06ed14000?, 0x696533?})                                                                                                                                                         github.com/openshift/must-gather-clean/pkg/obfuscator/regex.go:21 +0x1e                                                                                                                                                                                              github.com/openshift/must-gather-clean/pkg/obfuscator.(*targetObfuscator).Contents(0x0?, {0xc06ed14000?, 0xc001a6dbd0?})                                                                                                                                                             github.com/openshift/must-gather-clean/pkg/obfuscator/target_obfuscator.go:19 +0x5f                                                                                                                                                                                  github.com/openshift/must-gather-clean/pkg/obfuscator.(*MultiObfuscator).Contents(0xc07890a000?, {0xc07890a000?, 0x1a4006e9918?})
        github.com/openshift/must-gather-clean/pkg/obfuscator/multi_obfuscator.go:17 +0x42
github.com/openshift/must-gather-clean/pkg/cleaner.(*ContentObfuscator).ObfuscateReader(0xc00037ec30, {0x6e9b78?, 0xc0031a8010?}, {0x6e9bb8?, 0xc0031a8018?})
        github.com/openshift/must-gather-clean/pkg/cleaner/cleaning.go:195 +0x2fa
github.com/openshift/must-gather-clean/pkg/cleaner.(*FileContentObfuscator).ObfuscateFile(0xc00037ec30, {0xc0240c0fe6?, 0x6e9918?}, {0xc00058c500, 0xf7})
        github.com/openshift/must-gather-clean/pkg/cleaner/cleaning.go:142 +0x4f9
github.com/openshift/must-gather-clean/pkg/cleaner.(*FileProcessor).Process(0xc00037ec30, {0xc0240c0fe6, 0xf8})
        github.com/openshift/must-gather-clean/pkg/cleaner/cleaning.go:95 +0x14a
github.com/openshift/must-gather-clean/pkg/traversal.(*Worker).ProcessQueue(0xc0002bca68, 0x0?, 0x0?)
        github.com/openshift/must-gather-clean/pkg/traversal/worker.go:40 +0x192
github.com/openshift/must-gather-clean/pkg/traversal.(*FileWalker).Traverse.func1(0x0?, 0x0?, 0x0?)
        github.com/openshift/must-gather-clean/pkg/traversal/traversal.go:33 +0x42
created by github.com/openshift/must-gather-clean/pkg/traversal.(*FileWalker).Traverse
        github.com/openshift/must-gather-clean/pkg/traversal/traversal.go:32 +0xd2
```
